/*
 * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
 * SPDX-License-Identifier: Apache-2.0
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
syntax = "proto2";

import "google/protobuf/any.proto";

message Topic {
  optional string id = 1;
}

message Message {
  optional google.protobuf.Any payload = 1;
}

message SubscribeRequest {
  repeated Topic topics = 1;
}

message SubscribeDelivery {
  optional Topic topic = 1;
  optional Message message = 2;
}

message PublishRequest {
  optional Topic topic = 1;
  optional Message message = 2;
}

message PublishResponse {}

service MessageBroker {
  rpc SubscriptionStream(SubscribeRequest) returns (stream SubscribeDelivery) {}
  rpc Publish(PublishRequest) returns (PublishResponse) {}
}
